broker/commands.py

```
@@ -58,14 +47,11 @@ def checkin(vm, all_):
     """
     inventory = helpers.load_inventory()
     to_remove = []
-    for num, host in enumerate(inventory):
-        if str(num) in vm or host["hostname"] in vm or all_:
-            to_remove.append((num, host))
-    # reconstruct the hosts and call their release methhod
-    for num, host in to_remove[::-1]:
-        del inventory[num]
-        logger.info(f"Checking in {host['hostname']}")
-    helpers.update_inventory(inventory, replace_all=True)
+    for num, host_export in enumerate(inventory):
+        if str(num) in vm or host_export["hostname"] in vm or all_:
+            to_remove.append(VMBroker.reconstruct_host(host_export))
```

Do you think we need to raise here info message in case of invalid input VM's?

---

The way this is written, you can supply any number of bad inputs and it simply won't act on them